walnut/converter/converter.go

```diff
@@ -0,0 +1,156 @@
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package converter
+
+import (
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+	"os"
+
+	"github.com/AliceO2Group/Control/common"
+	"github.com/AliceO2Group/Control/common/controlmode"
+	"github.com/AliceO2Group/Control/common/gera"
+	"github.com/AliceO2Group/Control/core/task"
+	"github.com/AliceO2Group/Control/core/task/channel"
+)
+
+// return pointer to float64
+func createFloat(x float64) *float64 {
+	return &x
+}
+
+// return pointer to bool
+func createBool(x bool) *bool {
+	return &x
+}
+
+// return pointer to string
+func createString(x string) *string {
+	return &x
+}
+
+// ExtractTaskClasses takes in a DPL Dump string and extracts
+// an array of Tasks
+func ExtractTaskClasses(DPL Dump) (tasks []*task.Class, err error) {
+
+	for index := range DPL.Workflows {
+		var channelName string
+		if index+1 == len(DPL.Workflows) {
+			channelName = DPL.Workflows[index].Name
+		} else {
+			channelName = "from_" + DPL.Workflows[index].Name + "_to_" + DPL.Workflows[index+1].Name
+		}
+
+		taskName := DPL.Workflows[index].Name
+		defaultBindChannel := channel.Inbound{
+			Channel: channel.Channel{
+				Name:        channelName,
+				Type:        channel.ChannelType("push"), // defaulting to push
+				SndBufSize:  1000,
+				RcvBufSize:  1000,
+				RateLogging: 60,
+				Transport:   channel.TransportType("shmem"),
+			},
+			Addressing: "ipc",
+		}
+
+		// Not required for Task Templates
+		/*
+			defaultConnectChannel := channel.Outbound{
+				Channel: channel.Channel{
+					Name:        workflowName,
+					Type:        channel.ChannelType(""),
+					SndBufSize:  1000,
+					RcvBufSize:  1000,
+					RateLogging: 60,
+					Transport:   channel.TransportType("shmem"),
+				},
+				// Target: "", No default value
+			}
+		*/
+
+		task := task.Class{
+			Identifier: task.TaskClassIdentifier{
+				Name: taskName,
+			},
+			Defaults: gera.MakeStringMapWithMap(map[string]string{
+				"user": "flp",
+			}),
+			Control: struct {
+				Mode controlmode.ControlMode "yaml:\"mode\""
+			}{Mode: controlmode.FAIRMQ},
+			Command: &common.CommandInfo{
+				// Env: []string, -> Default to empty array
+				Shell:     createBool(true),
+				Value:     &DPL.Metadatas[index+1].Executable,
+				Arguments: DPL.Metadatas[index+1].CmdlLineArgs,
+				User:      createString("flp"),
+			},
+			Wants: task.ResourceWants{
+				Cpu:    createFloat(0.15),
+				Memory: createFloat(128),
+				Ports:  task.Ranges{}, //begin - end OR range
+			},
+			Bind: []channel.Inbound{defaultBindChannel},
+			Properties: gera.MakeStringMapWithMap(map[string]string{
+				"severity": "trace",
+				"color":    "false",
+			}),
+			// Connect:   []channel.Outbound{defaultConnectChannel},
+		}
+		// fmt.Printf("\nTASK:\n%v\n", task)
```

Comment:
I propose using `common/logger` and related `logrus` facilities instead of `fmt` for logging output. See https://github.com/AliceO2Group/Control/blob/master/coconut/control/control.go for some examples.

If you ever need to dump a complex struct for debugging purposes, you can also use `k0kubun/pp`. Examples can be found here https://github.com/AliceO2Group/Control/blob/master/core/workflow/load.go and elsewhere in AliECS.

Not urgent & can be a JIRA ticket for you.

Reply:
Issue created: https://alice.its.cern.ch/jira/browse/OCTRL-246